presto-main/src/main/java/com/facebook/presto/operator/DeleteOperator.java

```diff
@@ -144,6 +151,9 @@ public Page getOutput()
 
         Collection<Slice> fragments = pageSource().commit();
         committed = true;
+        long usedMemoryBytes = pageSource().getUsedMemoryBytes();
```

Comment:
This pattern seems to repeat in all the operators. How about having a class with methods `setMemoryUsage(bytes)` and `freeMemoryUsage()`